src/application/snapshot/utils.ts

```diff
@@ -1,36 +1,28 @@
-/* eslint-disable @typescript-eslint/restrict-template-expressions */
-
 import { SNAPSHOT_WEBSITE_URL } from './constants';
-import { GetProposalsOptions, ProposalData } from './types';
+import { ProposalData } from './types';
 
-export const generateGetProposalsQuery = (
-  snapshotNames: string[],
-  options: GetProposalsOptions,
-) => {
-  // TODO: use GQL variables for this
-  return `query Proposals {
-  proposals (
-    first: ${options.first},
-    skip: 0,
-    where: {
-      space_in: [${snapshotNames.map((name) => {
-        return `"${name}"`;
-      })}],
-      state: "active"
-    },
-    orderBy: "end",
-    orderDirection: asc
-  ) {
-    id
-    title
-    body
-    end
-    author
-    space {
-      id
-    }
-  }
-}`;
+export const generateGetProposalsQuery = () => {
+  return `query Proposals($first: Int,$skip: Int , $snapshotNames: [String!]) {
+      proposals (
+        first: $first,
+        skip: $skip,
+        where: {
+          space_in: $snapshotNames,
+          state: "active"
+        },
+        orderBy: "end",
+        orderDirection: asc
+      ) {
+        id
+        title
+        body
+        end
+        author
+        space {
+          id
+        }
+      }
+    }`;
 };
```

Comment:
after refactoring we only pass a single snapshotName, can you change argument type to `String!` and rename to `$snapshotName` to highlight that it should be a single snapshot passed?

Reply:
btw. that's interesting that it was fine for GQL that you were passing a string instead of array of strings

Reply:
Yeah, quite weird, but actuially we need to pass [String!], otherwise we are getting error: 
![image](https://github.com/idriss-crypto/browser-extensions/assets/106707979/1b950fe5-765b-4fc3-b82d-afef84422bc6)

To make it consistent I've changed the payload type
![image](https://github.com/idriss-crypto/browser-extensions/assets/106707979/5e863453-573e-4915-8c25-c95678101f42)

Reply:
ok thanks 😄